### ISSUE TITLE
 DeprecationWarning: Calling an asynchronous function without callback is deprecated.

### ISSUE THREAD

**Issue by fan1200** [f-hj/slsk-client#17]
Hi,

Been working with this library for a few weeks and it works great. Sadly there is a weird problem with the code. When running I get this exception on certain files. It eventually results into a "peer error".

> (node:5580) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
> UNCAUGHT EXCEPTION - keeping process alive: { Error: This socket has been ended by the other party
>     at Socket.writeAfterFIN [as write] (net.js:364:12)
>     at Timeout.setTimeout [as _onTimeout] (\slsk-client\lib\download-peer-file.js:25:14)
>     at ontimeout (timers.js:498:11)
>     at tryOnTimeout (timers.js:323:5)
>     at Timer.listOnTimeout (timers.js:290:5) code: 'EPIPE' }

I use this on a Windows machine with node version v8.11.4

An example item which fails
```json
{ "user": "USER",
  "file": "l:\\music one\\absolute music - 320\\FILE.mp3",
  "size": 9051058,
  "slots": true,
  "bitrate": 320,
  "speed": 72159 }
```

**Comment by sortofdev** [comment]
Running into similar issue a lot recently, output sample with debug enabled:
```
  slsk:peer:i Directly allowed. Connecting to peer with PeerInit + 1c6b3c56 +0ms
  slsk:peer:file downloadPeerFile SomeUser +0ms
  slsk:peer:file noPierce SomeUser connected +262ms
  slsk:peer:file file socket close SomeUser +270ms
  slsk:peer:file ERROR: token 45070000 not exist +1ms
(node:5608) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
  slsk:peer:file noPierce sending 8* 00 +731ms
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: This socket has been ended by the other party
    at Socket.writeAfterFIN [as write] (net.js:351:12)
    at Timeout.setTimeout [as _onTimeout] (slsk-client\lib\download-peer-file.js:25:14)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
    at Timer.listOnTimeout (timers.js:270:5)

```

**Comment by f-hj** [comment]
Should be fixed in 1.1.0
